### ISSUE TITLE
Component::printInputInfo() should print "<no connectees>" if there are no connectees

### ISSUE THREAD

**Issue by chrisdembia** [opensim-org/opensim-core#1618]
Currently, the output of `Component::printInputInfo()` may look like the following:

```
Inputs for component controller of type [ToyPropMyoController] along with connectee names:
    [double] activation : 
```

Since `activation` has no connectees, no connectee names are listed. It'd be neat if we could print `<no connectees>` after `activation`.

EDIT: the code *does* try to print `no connectees`, but `getNumConnectees()` seems to be giving the wrong information. I think the issue occurs for single-value Inputs, which always have 1 connectee, even if it isn't specified.

**Comment by tkuchida** [comment]
Won't [this line](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/Component.cpp#L1534) print "no connectees" if `input->getNumConnectees() == 0`? Are you sure there isn't a connectee whose name is null?

**Comment by chrisdembia** [comment]
Yeah I made a mistake; I edited the original post.

**Comment by tkuchida** [comment]
> I think the issue occurs for single-value Inputs, which always have 1 connectee, even if it isn't specified.

Perhaps add a check for `isConnected()`?
```cpp
if ( input->getNumConnectees() == 0 ||
     (input->getNumConnecteed() == 1 && !input->isConnected()) )
```

**Comment by chrisdembia** [comment]
That should work.

**Comment by tkuchida** [comment]
Fixed by #1640.
